src/main/java/stirling/software/SPDF/config/security/InitialSecuritySetup.java

```diff
@@ -30,6 +34,20 @@ public void init() {
             String initialPassword =
                     applicationProperties.getSecurity().getInitialLogin().getPassword();
             if (initialUsername != null && initialPassword != null) {
+                try {
+                    // https://github.com/Stirling-Tools/Stirling-PDF/issues/976
+                    userService.isUsernameValidWithReturn(initialUsername);
+                } catch (IllegalArgumentException e) {
+                    Path pathToFile = Paths.get("configs/settings.yml");
+                    try {
+                        if (Files.exists(pathToFile)) {
+                            Files.delete(pathToFile);
```

Comment:
I dont think if a user enters a incorrect username we should just wipe all settings they entered, i would prefer just a error message and stopping the app completly

Reply:
This is the first setup if there are no users yet. Therefore I assumed that the file would be created fresh.

Reply:
I still don't think it's the right way and they might have configured other settings too etc

Reply:
https://github.com/Stirling-Tools/Stirling-PDF/blob/f2b7aeeb1cf4ec219383063edc5f78d40f18a3cf/src/main/java/stirling/software/SPDF/SPdfApplication.java#L72

The settings.yml is only read once, if it is edited afterwards because the username is not valid, it is ignored.
In Docker, you can rebuild the container and it works. But it doesn't work locally, or I don't know how to trigger it.

Reply:
id rather just leave the invalid username in file and have the user fix it themselves, we can just have the app throw error log and exit

Reply:
I think I'm expressing myself wrong. If it is started incorrectly configured, the app cannot be restarted afterwards. It's as if the settings.yml hasn't been changed.

Reply:
So i am imagining it being
- user edits settinsg yml and inputs wrong username
- starts app
- apps seeing wrong username, calls System.exit(1); or something similar completly stopping docker container
- user has to check logs to see error message, sees its wrong and checks file and edits
- restarts docker 

I know your way it will restore it to default values, but i dont want any other settings changes they made to also be lost

Reply:
I'll test it tomorrow with System.exit(1).

Reply:
The container handling is not the actual problem, but the local application 🫣But I'll see about that tomorrow.

Reply:
System exit should shut down the app itself